File: challenge/challenge32.java
import java.util.*;

class Challenge32{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the length of rows: ");
        int row = sc.nextInt();
        System.out.print("Enter the length of column: ");
        int col = sc.nextInt();

        for(int r = 0; r<row; r++){
            for(int c = 0; c<col; c++){
                if( (c == 0) || ((c == col-1) && (r != 0 && r != row-1) ) ){
                    System.out.print("* ");
                }else if( (r == 0 || r == row-1) && ( c > 0 && c < col-1) ){
                    System.out.print("* ");
                }else{
                    System.out.print("  ");
                }
            }
            System.out.println();
        }
    }
}

File: python solution/challenge32.py
row = int(input("Enter the length of row: "))
col = int(input("Enter the length of column: "))

for r in range(row):
    for c in range(col):
        if( (c == 0) or ((c == col-1) and ( r != 0 and r != row-1 )) ):
            print("*", end=" ")
        elif( (r == 0  or r == row-1) and ( c > 0 and c < col-1 ) ):
            print("*", end=" ")
        else:
            print(" ", end=" ")
    print()